opentelemetry/benches/baggage.rs

```diff
@@ -11,10 +11,10 @@ const MAX_KEY_VALUE_PAIRS: usize = 64;
 // Apple M4 Pro
 //     Total Number of Cores:	14 (10 performance and 4 efficiency)
 // Results:
-// set_baggage_static_key_value 12 ns
-// set_baggage_static_key 28 ns
-// set_baggage_dynamic 60 ns
-// set_baggage_dynamic_with_metadata 112 ns
+// set_baggage_static_key_value 14 ns
+// set_baggage_static_key 26 ns
+// set_baggage_dynamic 54 ns
+// set_baggage_dynamic_with_metadata 75 ns
```

Comment:
@cijothomas maybe you could re-run it on your side. I did it on a M3 Pro and it was mostly identical to the M4 Pro before the change, except `set_baggage_static_key_value` which was a bit worse and `set_baggage_dynamic_with_metadata` was quite better.

Reply:
I'll rename the PR to enable the performance regression test on the CI server -->
https://github.com/open-telemetry/opentelemetry-rust/blob/ddac8e1ef3d679ce932e768ecd73af19113ef2b8/.github/workflows/pr_criterion.yaml#L9

Reply:
From [the build run](https://github.com/open-telemetry/opentelemetry-rust/actions/runs/13856678788/job/38774968985?pr=2804) it's looking good:

| name                                         | baseDuration      | changesDuration    | difference |
|----------------------------------------------|-------------------|--------------------|------------|
| 'set_baggage_dynamic'                        | '79.7±1.08ns'     | '**75.9±1.01ns**'  | '-4.8'     |
| 'set_baggage_dynamic_with_metadata'          | '113.2±2.01ns'    | '**108.0±1.74ns**' | '-4.8'     |
| 'set_baggage_static_key'                     | '56.5±0.91ns'     | '**54.8±0.49ns**'  | '-2.9'     |
| 'set_baggage_static_key_value'               | '41.3±2.20ns'     | '**32.2±0.55ns**'  | '-22'      |